### ISSUE TITLE
to ensure easier use with data that comes in different formats, move merge of all input data to first bits of work.

### ISSUE THREAD

**Issue by JustinGOSSES** [JustinGOSSES/predictatops#16]
Currently, things like fining all available curves in wells and finding neighboring wells are done with just the input files needed for those tasks, then the results merged. 

If someone is bringing in geographic coordinate data that isn't in txt or CSV form, it might be easier to adapt this code if they only had to write their own imports and transforms at the very beginning. 

Some of the code having to do with SiteID <=> UWI <=> Well log filesnames, for example, is transformed a couple of different times in different early modules. 

In summary, It would probably make more sense to just do it once and have a big dataframe from then on. The intermediate file sizes would be bigger but the code would be less complex to adapt. If everything was a dataframe from check / load onwards.
